pkg/metrics/metrics.go

```diff
@@ -0,0 +1,126 @@
+package metrics
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog/v2"
+
+	"github.com/blang/semver"
+	gensharev1alpha1 "github.com/openshift/csi-driver-shared-resource/pkg/generated/listers/sharedresource/v1alpha1"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	separator        = "_"
+	sharesSubsystem  = "openshift_csi_shares"
+	sharesCountQuery = sharesSubsystem + separator + "total"
+
+	MetricsPort = 6000
+)
+
+var (
+	sharesCountDesc = prometheus.NewDesc(
+		sharesCountQuery,
+		"Counts shares by kind of resource and namespace",
+		[]string{"source_namespace", "kind"},
```

Comment:
with the recent breaking out of the shared crd to sharedConfigMap and sharedSecret crd, we should have different metrics for eaach of those new types now, and hence you lose the "kind" label @alicerum 

I would also expect a "source_name" label

Of course adjust the descriptions accordingly

I think it is also important to explain what source ns/name mean .... otherwise it is natural for consumers to get confused by their typical understanding of what those labels mean with other metrics, as @jan--f  did

Reply:
@gabemontero i have number of mounts prepared as a different PR, wanted to push them separately. I think I had this discussion with @adambkaplan , that I should create PR as soon as I had one metrics implemented.
I'll separate the metrics for config maps and secrets. We had a discussion with @adambkaplan today, that I might want to move all this to https://github.com/openshift/csi-driver-shared-resource-operator due to a number of reasons, including controller running in a daemon set (on every node). What do you think?